### PR TITLE
[13] create button component

### LINKED ISSUE #13: Create button component
Design: https://www.figma.com/file/84q6WTlMuI3NZWEWo0tGYa/Nowa-strona-CC-dev?node-id=903%3A2120

AC:
- Component added to storybook
- Component tested with vitest

### PR DESCRIPTION
**Pull request by n0macx**
Closes #13 

Information:

- [x] Do your changes affect most of the application?
  - [x] Did you describe it in the task?
  - [x] Did you inform others about it on Slack?
- [ ] Have you installed a new library?
  - [ ] Have you consulted it with the others?

Checklist:

- [x] Does the Pull Request title meet the rules? (eg. `[9] Add name's rules for PR to readme`)
- [x] Does the name of your branch meet the rules? (eg. `9-add-names-rules-for-pr-to-readme`)
- [x] Does the Pull Request description contain `Closes #<issue number>` (eg. `Closes #9`)?
- [x] Did Husky run all jobs? (you didn't use the `no-verify` flag)
- [x] Did you describe the task in the linked issue? (AC, useful links, description)
- [x] Did you link the issue to the pull request through the `Development` tab in the pull request options?

Remainders:

- Make sure your commit messages contain useful information
- Make sure to use pnpm and do not push other lock files

### CODE REVIEW

File: package.json
Comment (jedrzejratajczak) [inline package.json:23]: Remove this package as we install it globally in our OS

File: prettier.config.js
Comment (jedrzejratajczak) [inline prettier.config.js:4]: Make sure to first merge [this branch](https://github.com/CodersCrew/coderscrew-website/pull/38) if you want to use this lib here

**Review by jedrzejratajczak — CHANGES_REQUESTED** [review]
Delete package-lock.json, please read the PR template and our contribution guide.
Make sure Vercel check passes.
If you check the box in the PR template and it has sub checkboxes they should also be completed.

File: src/pages/index.tsx
Comment (jedrzejratajczak) [inline src/pages/index.tsx:11]: We use Storybook for creating basic components. Please do not change the content of this file as it is unnecessary.

File: package.json
Comment (n0macx) [reply package.json:23]: Removed. Thanks for the comment.

File: prettier.config.js
Comment (n0macx) [reply prettier.config.js]: Waiting for approval.

File: src/pages/index.tsx
Comment (n0macx) [reply src/pages/index.tsx]: Done. Thanks for the comment.

**Comment by n0macx** [comment]
> Delete package-lock.json, please read the PR template and our contribution guide. Make sure Vercel check passes. If you check the box in the PR template and it has sub checkboxes they should also be completed.

I have a problem with logging to vercel could You resend me an invitation please ?

File: package.json
Comment (jedrzejratajczak) [inline package.json:23]: Still to remove and as we won't install any new lib the `pnpm-lock.yaml` file should remain unchanged.

File: .prettierrc.js
Comment (jedrzejratajczak) [inline .prettierrc.js:4]: You can delete this file

File: src/components/Button.tsx
Comment (jedrzejratajczak) [inline src/components/Button.tsx:7]: I would add padding so it will be closer to the design. The secondary button looks a bit off.

File: src/components/Button.tsx
Comment (jedrzejratajczak) [inline src/components/Button.tsx:9]: This should be something received by props. Why would we use the button to log `clicked` string in the console?

File: package.json
Comment (n0macx) [reply package.json:23]: Dona. Thanks for the comment.

File: .prettierrc.js
Comment (n0macx) [reply .prettierrc.js:4]: Done

File: src/components/Button.tsx
Comment (n0macx) [reply src/components/Button.tsx:9]: Done. Thanks for the review

File: src/components/Button.tsx
Comment (n0macx) [reply src/components/Button.tsx]: Done. Thanks for the comment.

File: src/components/Button/Button.test.tsx
Comment (jedrzejratajczak) [inline src/components/Button/Button.test.tsx:2]: we don't need to import from vitest

File: src/components/Button/Button.tsx
Comment (jedrzejratajczak) [inline src/components/Button/Button.tsx:19]: to props

File: src/components/Button/Button.tsx
Comment (jedrzejratajczak) [inline src/components/Button/Button.tsx:9]: not optional

File: src/components/Button/Button.tsx
Comment (n0macx) [reply src/components/Button/Button.tsx]: Done

File: src/components/Button/Button.tsx
Comment (n0macx) [reply src/components/Button/Button.tsx]: Done

**Review by jedrzejratajczak — APPROVED** [review]

File: src/components/Button/button.test.tsx
Comment (KlaraGajaszek) [inline src/components/Button/button.test.tsx:8]: - if you write const {getByText} = render(<Button type="button" variant="primary" label="Primary" onClick={() => {}} />) then we will have less code
- type="button"  label="Primary" onClick={() => {}} is default/basic props so you can hand only variant and the other props will have one declaration

**Review by KlaraGajaszek — CHANGES_REQUESTED** [review]

**Pull request merged**
